src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/SheetWriter.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.befta.dse.ccd.definition.converter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+
+public class SheetWriter {
+
+    private ArrayList<String> keys;
+    private static CellStyle cellDateStyle;
+
+    public XSSFWorkbook addSheetToXlxs(XSSFWorkbook workbook, String sheetName, ArrayNode sheetArrayNode) {
+        cellDateStyle = ExcelDateUtils.getExcelDateCellStyle(workbook);
+        if (sheetArrayNode.size() > 0){
+            Sheet sheet = workbook.createSheet(sheetName);
+            sheet.createRow(0).createCell(0).setCellValue(sheetName);
+            sheet.createRow(1); //blank row required - normally contains description
+            createHeaderRow(sheet, sheetArrayNode);
+
+            Iterator iterator = sheetArrayNode.iterator();
+            while (iterator.hasNext()) {
+                JsonNode row = (JsonNode) iterator.next();
+                writeRowToSheet(sheet, row);
+            }
+
+        }
+
+        return workbook;
+
+    }
+
+    /**
+     * Create a row consisting of the column headers for the particular ccd definition sheet/tab
+     * @param sheet
+     * @param sheetArrayNode
+     */
+    private void createHeaderRow(Sheet sheet, ArrayNode sheetArrayNode) {
+        Iterator<String> headers = sheetArrayNode.get(0).fieldNames();
+        ArrayList<String> keys = new ArrayList<>();
+
+        int nextRow = sheet.getPhysicalNumberOfRows();
+        Row row = sheet.createRow(nextRow);
+        int columnIndex = 0;
+        while (headers.hasNext()){
+            Cell cell = row.createCell(columnIndex);
+            String headerName = headers.next();
+            cell.setCellValue(headerName);
+            keys.add(headerName);
+            columnIndex++;
+        }
+
+        this.keys = keys;
```

Comment:
calculating and setting the value of a global variable as an additional activity is not good.
The key list can be calculated by a separate method and its return value can be used here.
Or at least the `keys` can be returned from the method and the global variable where needed.

Reply:
done